packages/@aws-cdk/aws-iot-actions/test/cloudwatch/cloudwatch-alarm-action.test.ts

```diff
@@ -0,0 +1,156 @@
+import { CloudWatchAlarmAction, CloudWatchAlarmActionProps } from '../../lib/cloudwatch-alarm-action';
+
+test('Default cloudwatch alarm action', () => {
+  // Given
+  const stack = new cdk.Stack();
+  const topicRule = new iot.TopicRule(stack, 'MyTopicRule', {
+    sql: iot.IotSql.fromStringAsVer20160323("SELECT topic(2) as device_id, stateReason, stateValue FROM 'device/+/data'"),
+  });
+  const alarmArn = 'arn:aws:cloudwatch:us-east-1:123456789012:alarm:MyAlarm';
+  const alarm = cloudwatch.Alarm.fromAlarmArn(stack, 'MyAlarm', alarmArn);
+  const cloudWatchAlarmActionProps: CloudWatchAlarmActionProps = {
+    stateReason: '${stateReason}',
+    stateValue: '${stateValue}',
+  };
+
+  // When
+  topicRule.addAction(new CloudWatchAlarmAction(alarm, cloudWatchAlarmActionProps));
+
+  // Then
+  Template.fromStack(stack).hasResourceProperties('AWS::IoT::TopicRule', {
+    TopicRulePayload: {
+      Actions: [
+        {
+          CloudwatchAlarm: {
+            AlarmName: 'MyAlarm',
+            RoleArn: {
+              'Fn::GetAtt': ['MyTopicRuleTopicRuleActionRoleCE2D05DA', 'Arn'],
+            },
+            StateReason: '${stateReason}',
+            StateValue: '${stateValue}',
+          },
+        },
+      ],
+    },
+  });
+
+  Template.fromStack(stack).hasResourceProperties('AWS::IAM::Role', {
+    AssumeRolePolicyDocument: {
+      Statement: [
+        {
+          Action: 'sts:AssumeRole',
+          Effect: 'Allow',
+          Principal: {
+            Service: 'iot.amazonaws.com',
+          },
+        },
+      ],
+      Version: '2012-10-17',
+    },
+  });
+
+  Template.fromStack(stack).hasResourceProperties('AWS::IAM::Policy', {
+    PolicyDocument: {
+      Statement: [
+        {
+          Action: 'cloudwatch:SetAlarmState',
+          Effect: 'Allow',
+          Resource: alarmArn,
+        },
+      ],
+      Version: '2012-10-17',
+    },
+    PolicyName: 'MyTopicRuleTopicRuleActionRoleDefaultPolicy54A701F7',
+    Roles: [{ Ref: 'MyTopicRuleTopicRuleActionRoleCE2D05DA' }],
+  });
+});
+
+test('can set stateReason', () => {
+  // Given
+  const stack = new cdk.Stack();
+  const topicRule = new iot.TopicRule(stack, 'MyTopicRule', {
+    sql: iot.IotSql.fromStringAsVer20160323("SELECT topic(2) as device_id, stateReason, stateValue FROM 'device/+/data'"),
+  });
+  const alarmArn = 'arn:aws:cloudwatch:us-east-1:123456789012:alarm:MyAlarm';
+  const alarm = cloudwatch.Alarm.fromAlarmArn(stack, 'MyAlarm', alarmArn);
+  const cloudWatchAlarmActionProps: CloudWatchAlarmActionProps = {
+    stateReason: 'Test SetAlarmState',
+    stateValue: '${stateValue}',
+  };
+
+  // When
+  topicRule.addAction(new CloudWatchAlarmAction(alarm, cloudWatchAlarmActionProps));
+
+  // Then
+  Template.fromStack(stack).hasResourceProperties('AWS::IoT::TopicRule', {
+    TopicRulePayload: {
+      Actions: [
+        Match.objectLike({ CloudwatchAlarm: { StateReason: 'Test SetAlarmState' } }),
+      ],
+    },
+  });
+});
```

Comment:
I don't see the point of this test - it doesn't cover anything the previous one didn't.

Reply:
Removed